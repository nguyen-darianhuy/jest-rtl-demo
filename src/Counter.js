import React from "react";

const Counter = ({ startingNum = 0 }) => {
  const [num, setNum] = React.useState(startingNum);
  return (
    <div>
      <span style={{ paddingRight: 10 }} data-testid="number">
        {num}
      </span>
      <button
        onClick={() => {
          if (num < 5) {
            setNum((num) => num + 1);
          }
        }}
      >
        Increment
      </button>
    </div>
  );
};

export default Counter;
