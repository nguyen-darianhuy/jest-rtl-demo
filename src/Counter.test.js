import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Counter from "./counter.js";

describe("Counter", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("counter increments", () => {
    const { getByRole, getByTestId } = render(<Counter />);
    expect(getByTestId("number")).toHaveTextContent("0");
    fireEvent.click(getByRole("button"));
    expect(getByTestId("number")).toHaveTextContent("1");
  });

  test("counter caps out", () => {
    expect(false).toBe(false);
    const { getByRole, getByTestId } = render(<Counter startingNum={5} />);
    expect(getByTestId("number")).toHaveTextContent("5");
    fireEvent.click(getByRole("button"));
    expect(getByTestId("number")).toHaveTextContent("5");
  });
});
