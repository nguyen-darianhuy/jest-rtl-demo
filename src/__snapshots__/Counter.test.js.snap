// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <span
      data-testid="number"
      style="padding-right: 10px;"
    >
      0
    </span>
    <button>
      Increment
    </button>
  </div>
</DocumentFragment>
`;
